reference/build_helpers/autotools.rst

```diff
@@ -1,58 +1,160 @@
 .. _autotools_reference:
 
-
 AutoToolsBuildEnvironment
 =========================
 
 Prepares the needed environment variables to invoke  **configure**/**make**:
 
 .. code-block:: python
 
-   from conans import ConanFile, AutoToolsBuildEnvironment
+    from conans import ConanFile, AutoToolsBuildEnvironment
 
-   class ExampleConan(ConanFile):
-       ...
+    class ExampleConan(ConanFile):
+        ...
 
+        def build(self):
+            env_build = AutoToolsBuildEnvironment(self)
+            env_build.configure()
+            env_build.make()
 
-       def build(self):
-         env_build = AutoToolsBuildEnvironment(self)
-         env_build.configure()
-         env_build.make()
+Or calling `configure` and `make` manually:
 
+.. code-block:: python
 
+    from conans import ConanFile, AutoToolsBuildEnvironment
 
-Or calling `configure` and `make` manually:
+    class ExampleConan(ConanFile):
+        ...
+
+        def build(self):
+            env_build = AutoToolsBuildEnvironment(self)
+            with tools.environment_append(env_build.vars):
+                self.run("./configure")
+                self.run("make")
+
+Constructor
+-----------
+
+.. code-block:: python
+
+    class AutoToolsBuildEnvironment(object):
+
+        def __init__(self, conanfile)
+
+Parameters:
+    - **conanfile** (Required): Conanfile object. Usually ``self`` in a conanfile.py
+
+Attributes
+----------
+
+You can adjust the automatically filled values modifying the attributes like this:
 
 .. code-block:: python
+   :emphasize-lines: 8, 9, 10
+
+    from conans import ConanFile, AutoToolsBuildEnvironment
+
+    class ExampleConan(ConanFile):
+        ...
+
+        def build(self):
+            env_build = AutoToolsBuildEnvironment(self)
+            env_build.fpic = True
+            env_build.libs.append("pthread")
+            env_build.defines.append("NEW_DEFINE=23")
+
+            with tools.environment_append(env_build.vars):
+                self.run("./configure")
+                self.run("make")
+
+fpic
```

Comment:
I think for this I prefer the table style. More compact.

Reply:
I also think a table is more compact in this case as there is no much additional information. However, I think it is better to have a uniform style format

Reply:
If it is only for the build helpers I think it's ok. Other tables like the integrations section are different.